packages/mdc-textfield/_mixins.scss

```diff
@@ -250,15 +262,34 @@
     z-index: 2;
   }
 
-  // stylelint-disable plugin/selector-bem-pattern
-  .mdc-text-field__input:hover ~ .mdc-text-field__idle-outline {
-    border: 1px solid;
+  .mdc-notched-outline,
+  .mdc-notched-outline__idle {
+    @include mdc-text-field-outline-corner-radius($mdc-text-field-border-radius);
   }
+}
+
+@mixin mdc-text-field-outline-color_($color) {
+  @include mdc-notched-outline-idle-color($color);
+  @include mdc-notched-outline-color($color);
+}
+
+@mixin mdc-text-field-hover-outline-color_($color) {
+  &:not(.mdc-text-field--focused) {
+    .mdc-text-field__input:hover ~,
+    .mdc-text-field__icon:hover ~ {
+      @include mdc-notched-outline-idle-color($color);
+
+      .mdc-notched-outline {
```

Comment:
Put a bunch of comments here around the ~ and sibling selector, and why thats hella complicated because `mdc-notched-outline__path` is a sub-element but `mdc-notched-outline__idle` is a sibling element.

And then file a GitHub issue to investigate making `mdc-notched-outline__idle` a sub-element of `mdc-notched-outline`.